src/compositemark/boxplot.ts

```diff
@@ -165,86 +160,193 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT
+        continuousAxis = 'x';
+        continuousAxisChannelDef = encoding.x;
+        discreteAxisFieldDef = encoding.y;
+      }
+    }
 
-  if (continuousAxisChannelDef && continuousAxisChannelDef.aggregate) {
-    const {aggregate: aggregate, ...continuousAxisWithoutAggregate} = continuousAxisChannelDef;
-    if (aggregate !== BOXPLOT) {
-      throw new Error(`Continuous axis should not have customized aggregation function ${aggregate}`);
+    if (continuousAxisChannelDef && continuousAxisChannelDef.aggregate) {
+      const {aggregate: aggregate, ...continuousAxisWithoutAggregate} = continuousAxisChannelDef;
+      if (aggregate !== BOXPLOT) {
+        throw new Error(`Continuous axis should not have customized aggregation function ${aggregate}`);
+      }
+      continuousAxisChannelDef = continuousAxisWithoutAggregate;
     }
-    continuousAxisChannelDef = continuousAxisWithoutAggregate;
   }
 
   return {
     discreteAxisFieldDef: discreteAxisFieldDef,
     continuousAxisChannelDef: continuousAxisChannelDef,
     discreteAxis: discreteAxis,
-    continuousAxis: continuousAxis
+    continuousAxis: continuousAxis,
+    is1D: !is2D
   };
 }
+
+export function boxTransform(encoding: Encoding<Field>, discreteAxisFieldDef: PositionFieldDef<Field>, continuousAxisChannelDef: PositionFieldDef<Field>, kIQRScalar: 'min-max' | number, is1D: boolean) {
+  const isMinMax = kIQRScalar === undefined;
+  const groupbyDef = boxTransformGroupby(discreteAxisFieldDef, encoding);
+
+  let transformDef:any = [
+      {
+        summarize: [
+          {
+            aggregate: 'q1',
+            field: continuousAxisChannelDef.field,
+            as: 'lowerBox'
+          },
+          {
+            aggregate: 'q3',
+            field: continuousAxisChannelDef.field,
+            as: 'upperBox'
+          },
+          {
+            aggregate: 'median',
+            field: continuousAxisChannelDef.field,
+            as: 'midBox'
+          }
+        ]
+      }
+  ];
+
+  if (isMinMax) {
+    transformDef[0].summarize = transformDef[0].summarize.concat([
+      {
+        aggregate: 'min',
+        field: continuousAxisChannelDef.field,
+        as: 'lowerWhisker'
+      },
+      {
+        aggregate: 'max',
+        field: continuousAxisChannelDef.field,
+        as: 'upperWhisker'
+      }
+    ]);
+  } else {
+    transformDef = transformDef.concat([
+      {
+        calculate: 'datum.upperBox - datum.lowerBox',
+        as: 'IQR'
+      },
+      {
+        calculate: 'datum.lowerBox - datum.IQR * ' + kIQRScalar,
+        as: 'lowerWhisker'
+      },
+      {
+        calculate: 'datum.upperBox + datum.IQR * ' + kIQRScalar,
+        as: 'lowerWhisker'
+      }
+    ]);
+  }
+
+  if (groupbyDef !== undefined) {
+    transformDef[0].groupby = groupbyDef;
+  }
+
+  return transformDef;
+}
+
+export function boxTransformGroupby(discreteAxisFieldDef: PositionFieldDef<Field>, encoding: Encoding<Field>): Array<Field | string> {
```

Comment:
This method only handle groupBy but doesn't add additional nodes that need more aggregation.  (For example, if color/size encodes aggregate value, you still need to add them to `summarize`.)